lxd/main_init.go

```diff
@@ -428,39 +457,22 @@ they otherwise would.
 		}
 	}
 
-	if imagesAutoUpdate {
-		ss, _, err := c.GetServer()
-		if err != nil {
-			return err
-		}
-
-		if val, ok := ss.Config["images.auto_update_interval"]; ok && val == "0" {
-			err = cmd.setServerConfig(c, "images.auto_update_interval", "")
-			if err != nil {
-				return err
-			}
-		}
-	} else {
-		err = cmd.setServerConfig(c, "images.auto_update_interval", "0")
-		if err != nil {
-			return err
-		}
-	}
+	data := &cmdInitData{}
+	data.Config = map[string]interface{}{}
 
 	if networkAddress != "" {
-		err = cmd.setServerConfig(c, "core.https_address", fmt.Sprintf("%s:%d", networkAddress, networkPort))
-		if err != nil {
-			return err
-		}
-
+		data.Config["core.https_address"] = fmt.Sprintf("%s:%d", networkAddress, networkPort)
 		if trustPassword != "" {
-			err = cmd.setServerConfig(c, "core.trust_password", trustPassword)
-			if err != nil {
-				return err
-			}
+			data.Config["core.trust_password"] = trustPassword
 		}
 	}
 
+	if imagesAutoUpdate {
+		data.Config["images.auto_update_interval"] = ""
```

Comment:
I went ahead and made the change, since it seems reasonable. Now essentially the old logic of interactive/auto is back to where it was, and the low-level code (used also by preseed) will allow anything.